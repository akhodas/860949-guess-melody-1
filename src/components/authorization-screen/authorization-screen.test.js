import React from 'react';
import renderer from 'react-test-renderer';

import AuthorizationScreen from './authorization-screen.jsx';

describe(`AuthorizationScreen`, () => {
  it(`correctly rendered`, () => {
    const tree = renderer
      .create(<AuthorizationScreen
        name={`name`}
        password={`password`}
        logIn={jest.fn()}
        onChange={jest.fn()}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
