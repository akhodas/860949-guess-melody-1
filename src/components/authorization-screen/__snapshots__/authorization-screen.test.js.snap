// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationScreen correctly rendered 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Необходима авторизация
  </h2>
  <p
    className="login__text"
  >
    Представтесь!
  </p>
  <form
    action=""
    className="login__form"
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="name"
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        onChange={[MockFunction]}
        type="text"
        value="password"
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      onClick={[MockFunction]}
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
